content/browser/web_contents/web_contents_impl.cc

```diff
@@ -9195,6 +9207,13 @@ bool WebContentsImpl::ContainsOrIsFocusedWebContents() {
   return false;
 }
 
+bool WebContentsImpl::ShouldRestrictAccessToFocusedFrame() {
+  // If this method is called on an inner WebContents, and the focused frame is
+  // outside of this WebContents's subtree, restrict the access.
+  return (GetOuterWebContents() || secure_embed_connector_) &&
+         !ContainsOrIsFocusedWebContents();
```

Comment:
The seems to work

```
return !ContainsOrIsFocusedWebContents();
```

Reply:
!ContainsOrIsFocusedWebContents() works in concept, if we make ContainsOrIsFocusedWebContents() work for outer WebContents when focus is inside secure embed (it currently doesn't work and we don't have to make it work until we support secure embed inside inner WebContents), and it would perform check unnecessarily when called on top WebContents. Checking on top WebContents should be the majority of the case.

Reply:
Added TODO. FWIW, there is a TODO on ContainsOrIsFocusedWebContents() already, and the caller probably should not call the expensive ContainsOrIsFocusedWebContents() directly for perf reason.